services/app-api/utils/validation/validation.ts

```diff
@@ -0,0 +1,85 @@
+import { nested, endDate } from "./schemas";
+import { schemaMap } from "./schemaMap";
+import { AnyObject } from "../types/types";
+import { VALIDATION_ERROR_MESSAGE } from "../constants/constants";
+
+// compare payload data against validation schema
+export const validateData = async (
+  validationSchema: AnyObject,
+  data: AnyObject,
+  options?: AnyObject
+) => {
+  try {
+    // returns valid data to be passed through API
+    return await validationSchema.validate(data, {
+      stripUnknown: true,
+      ...options,
+    });
+  } catch (e: any) {
+    console.log("VALIDATION ERRORS", e); // eslint-disable-line no-console
```

Comment:
Not sure if we want to keep these kinds of console logs going forward, so dropping a note here

Reply:
Removed. I had initially thought they might be good to keep in the console but they do make the tests noisier for kind of no reason, so 🤷